### ISSUE TITLE
[HOLD for payment 2024-06-06] [$500] Empty state image has too much overlap with message in attachment thread

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36166]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.38-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1707401686727289

## Action Performed:
1. Send a message with a vertical (portrait) image. Then create a thread on this image and view the thread.
## Expected Result: 
The empty state image/background pattern should not have so much overlap with the message.
## Actual Result: 
The empty state pattern has a LOT of overlap with the message, almost sitting directly behind it.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Screen Shot 2024-02-08 at 11 54 22 AM (2)](https://github.com/Expensify/App/assets/38435837/59340db7-4815-41a6-96f6-b33fac667af7)
With vertical tall image
![CleanShot 2024-02-08 at 09 13 00@2x (1)](https://github.com/Expensify/App/assets/38435837/3cba8557-708c-498b-abb9-2e01c9ba3a57)
with normal image
![CleanShot 2024-02-08 at 09 15 06@2x](https://github.com/Expensify/App/assets/38435837/68363295-e959-413a-ae7a-5fcb6381ae32)
without image
![CleanShot 2024-02-08 at 09 15 29@2x](https://github.com/Expensify/App/assets/38435837/9f7542a3-0bba-4789-b226-bfe408bf6953)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0180e51b2d8c0c04ec</li>
        <li>Upwork Job ID: 1755638484200894464</li>
        <li>Last Price Increase: 2024-04-24</li>
<li>Automatic offers: </li>
<ul>
<li>cubuspl42 | Reviewer | 0</li>
<li>rayane-djouah | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by shawnborton** [comment]
@rayane-djouah shared a rough mockup for you [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1707414825420289?thread_ts=1707412868.737769&cid=C01GTK53T8Q)

![image](https://github.com/Expensify/App/assets/2319350/fc561718-30f5-4ec3-880f-05748593375b)

**Comment by cubuspl42** [comment]
@shawnborton Does this require any extra approval, as it's a user-observable change? Or is your approval enough?

**Comment by shawnborton** [comment]
Got it, but I'm thinking not just thread reports but just regular DMs, #rooms, etc. Basically this part: 
![CleanShot 2024-02-09 at 12 47 48@2x](https://github.com/Expensify/App/assets/2319350/a2dc0a8f-17e2-4793-bed5-a4827393ee99)

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Image on background not in correct position

### What is the root cause of that problem?

We have a style applied here:
https://github.com/Expensify/App/blob/287613ea925964f66753892daaa3a779d61d0e2d/src/pages/home/report/ReportActionItemParentAction.tsx#L74

`styles.justifyContentEnd` is pushing the background image to the bottom. This only happen to the thread pages.

### What changes do you think we should make in order to solve the problem?

Just removing the `styles.justifyContentEnd` will fix the issue. 
```diff
-  <View style={[StyleUtils.getReportWelcomeContainerStyle(isSmallScreenWidth), styles.justifyContentEnd]}>
+  <View style={[StyleUtils.getReportWelcomeContainerStyle(isSmallScreenWidth)>
```

POC:

![image](https://github.com/Expensify/App/assets/12425932/73e38f56-f2c4-4fb1-91c0-3dd320e19ed3)

**Comment by shawnborton** [comment]
@rayane-djouah that feels good to me. @cubuspl42 @johncschuster I would love to have @rayane-djouah work on this as they've been super helpful in exploring solutions for this one.

**Comment by cubuspl42** [comment]
@shawnborton 

> maybe we should make this change across all room types (DM, #room, report, expense, etc) so that we get the consistent 60px overlap no matter where you are.

![image](https://github.com/Expensify/App/assets/2590174/d7be31b9-8496-46d1-9399-7c8bfc11decb)

Would you confirm the scope here?

**Comment by cubuspl42** [comment]
C+ reviewed 🎀 👀 🎀

I agree with @shawnborton that we should assign @rayane-djouah

**Comment by shawnborton** [comment]
> Would you confirm the scope here?

Haha, yes, let's fix this for ALL room/chat types.

**Comment by johncschuster** [comment]
Sounds great! I've just assigned @rayane-djouah 🎉

**Comment by miljakljajic** [comment]
Another similar example: https://github.com/Expensify/App/issues/35921

**Comment by miljakljajic** [comment]
If not, we can reopen. I just closed in reference to Shawn's comment on our issue.

**Comment by cubuspl42** [comment]
@rayane-djouah 

I asked John to confirm the scope right after you were assigned, see: https://github.com/Expensify/App/issues/36166#issuecomment-1941575002

We agreed to fix "Empty state image has too much overlap" for "ALL room/chat types". It's my mistake that I missed this during the PR review. Sorry about that!

> No, the solution to https://github.com/Expensify/App/issues/35921 is different

It doesn't matter; then our overall solution will have two parts. Please submit a follow-up PR with the second part of the solution! Thank you.

**Comment by shawnborton** [comment]
Before we confirm the expected height across desktop/mobile, I would love to know if we even think it's technically possible to make that part of the chat view not scrollable. 

If I were to do this in CSS, I think I would do something where I add the background empty state image to the very first message/item in the chat where it's basically absolutely positioned. And make sure the wrapper that wraps both the item's content and the background image doesn't have overflow: hidden. This way that first item always has the floating background image above it, but it's not counted as real estate that would eat up scrollable area in the entire chat view.

**Comment by cubuspl42** [comment]
@shawnborton I know it's a bit late, but I'm having doubts whether I completely understand these words:

> If I were to do this in CSS, I think I would do something where I add the background empty state image to the very first message/item in the chat where it's basically absolutely positioned. And make sure the wrapper that wraps both the item's content and the background image doesn't have overflow: hidden. This way that first item always has the floating background image above it, but it's not counted as real estate that would eat up scrollable area in the entire chat view.

While I share the basic intuition, I'm having trouble figuring out how this relates to _scrollable_ content. It makes things somewhat more complex.

You said you have a vision of how to do this in CSS. Could you provide an extremely simplified HTML/CSS mockup? I mean the one where an "empty state image" is a colored rectangle, and the "messages" (things inside the scrollable wrapper) are some lorem ipsum blocks with a border or something like that. It would help me review the PR so I could ensure that what was implemented is what you had in mind.

**Comment by shawnborton** [comment]
Totally, I can make a quick demo with HTML/CSS. Will try to get to it later this afternoon/tonight.

**Comment by shawnborton** [comment]
@cubuspl42 can you try this CodePen here? https://codepen.io/shawnborton/pen/vYMpzEL

Zoom in and out with your browser to make the content area bigger/smaller, which should show you how this thing works with overflow. 

Basically using the `&::before` selector and absolute positioning let's us add this red block to the yellow row, without making it add to the overall row height.
![CleanShot 2024-04-03 at 23 37 35@2x](https://github.com/Expensify/App/assets/2319350/528d8030-18d1-4532-98ad-e82cfcdf694b)

**Comment by cubuspl42** [comment]
Awesome!

I created [a React Native variant of this demo](https://github.com/cubuspl42/ReactNativeExpoMinimalApp/tree/BottomAlignedScrollView-1).

https://github.com/Expensify/App/assets/2590174/65aaca05-6e79-45fd-9e6b-21649b72a608

@shawnborton Would you review the video above? Does thing behave as they should?

@rayane-djouah Would you check out the code I shared? Does it help with understanding what we want to achieve layout-wise?

**Comment by shawnborton** [comment]
That is EXACTLY what I was thinking, looks great!

**Comment by mallenexpensify** [comment]
@allroundexperts reassigning, please take over as C+. If you don't have bandwidth, unassign yourself. Thanks PR is here:
- https://github.com/Expensify/App/pull/38449

**Comment by allroundexperts** [comment]
Thank you. I'll review this today & tomorrow.

**Comment by allroundexperts** [comment]
@mallenexpensify Can you please assign the related PR as well? I might miss it without the assignment.

**Comment by mallenexpensify** [comment]
Done, I also updated internal docs to inc. assigning the new C+ to the PR as a reviewer too.

**Comment by johncschuster** [comment]
[Bumped in Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1715804835747379)

**Comment by johncschuster** [comment]
@allroundexperts reminded me that they un-assigned themselves from this one and @eh2077 will review this. @eh2077, can we get your thoughts on [this](https://github.com/Expensify/App/pull/38449#issuecomment-2085343174) when you get a moment? Thank you!

**Comment by eh2077** [comment]
Yes, we're discussing [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1707237972842589)

**Comment by johncschuster** [comment]
It looks like we're waiting on Shawn to run another test build to see if this behavior persists. 👍

**Comment by johncschuster** [comment]
@yuwenmemon can you confirm if this was linked to a regression?

**Comment by shawnborton** [comment]
@rayane-djouah curious for your thoughts as well

**Comment by yuwenmemon** [comment]
Certainly, a regression from _somewhere_ - yes. Although yeah, maybe @rayane-djouah has better knowledge about _where_.

**Comment by yuwenmemon** [comment]
I don't think we need a regression test for this though.

**Comment by johncschuster** [comment]
# Payment Summary:
Contributor: @rayane-djouah - $500 - paid [via Upwork](https://www.upwork.com/nx/wm/offer/100932868) on June 6
Contributor+: @eh2077 - $500 - paid [via Upwork(need to create an offer for this)](https://www.upwork.com/nx/wm/offer/0) on June 6

**Comment by johncschuster** [comment]
All offers have been accepted. I'll issue payment on June 6 👍

**Comment by shawnborton** [comment]
I don't think we should be issuing payment just yet - I think we have had some regressions with this, right?

**Comment by eh2077** [comment]
@shawnborton Yeah, we should get regression fixed before issuing payment.

@rayane-djouah Can you take a look at this comment https://github.com/Expensify/App/issues/42795#issuecomment-2140678910 and fix the regression from this PR?

**Comment by johncschuster** [comment]
@shawnborton Ah yes, we shouldn't issue payment until the regression is fixed. I was commenting on auto-pilot 😅

**Comment by shawnborton** [comment]
Thoughts on @rayane-djouah's comment above @yuwenmemon?

@rayane-djouah - do you agree that [this bug](https://github.com/Expensify/App/issues/42760) was caused by this though?

**Comment by yuwenmemon** [comment]
@shawnborton Yeah it's 2 separate problems as @stitesExpensify mentioned [here](https://github.com/Expensify/App/issues/42795#issuecomment-2140678910). Certainly, the single expense displaying twice is an issue and not directly from here, but the UI issue still stands.

**Comment by yuwenmemon** [comment]
Yes, correct. There's a regression.

**Comment by johncschuster** [comment]
Thanks! I've updated the [payment summary](https://github.com/Expensify/App/issues/36166#issuecomment-2145270720) above to indicate that.

**Comment by eh2077** [comment]
> > @shawnborton Yeah, we should get regression fixed before issuing payment.
> > @rayane-djouah Can you take a look at this comment [#42795 (comment)](https://github.com/Expensify/App/issues/42795#issuecomment-2140678910) and fix the regression from this PR?
> 
> I don't think that #42795 is caused by my PR. I think the root cause of [that bug](https://github.com/Expensify/App/issues/42795) is we are displaying a single expense twice. The empty state image is designed to be displayed in the first action of a report. if we display two report actions with the image, one of them will be covered by the image. I think that the duplicate expense action bug is what need to fix in #42795. Do you agree?

As @rayane-djouah seems to be OOO, let me try to continue with the discussion. I also think the UI issue of https://github.com/Expensify/App/issues/42795 shouldn’t be a regression from this PR if following is true

_**From design perspective, we don't want to show two or more money requests in the `MoneyRequestView` (no matter if the request is editable or not), right?**_

This is the bug screenshot - the expense is displayed twice
<img width="1440" alt="image" src="https://github.com/Expensify/App/assets/117511920/207bc0fc-6042-4dbe-9d47-87903473dc01">

The normal ones look like
This is editable 
<img width="1067" alt="image" src="https://github.com/Expensify/App/assets/117511920/d6893b8b-9c20-467b-8c81-03e58b6de897">
This is not editable
<img width="1068" alt="image" src="https://github.com/Expensify/App/assets/117511920/a9a61fdb-0e7e-4290-9a9f-4821b4896195">

@shawnborton Can you help to drive clarity here? It'll also be helpful for the other issue https://github.com/Expensify/App/issues/42795

cc @yuwenmemon @stitesExpensify

**Comment by shawnborton** [comment]
Correct, we definitely do not want to display the expense information twice like that.

**Comment by eh2077** [comment]
@shawnborton  Even if they are different expenses, we also don't want to display more than one expense in this page, right?

**Comment by shawnborton** [comment]
If we have multiple expenses in the same report, we use a different view where we show expense preview cards, not the full expense details view:

![CleanShot 2024-06-11 at 18 52 22@2x](https://github.com/Expensify/App/assets/2319350/417c5519-80d2-48e6-b7e5-8c4f19acf7af)

**Comment by eh2077** [comment]
I see. Then clicking a preview card will only show details of an expense. If so, I think the implementation of this PR should be fine and it shouldn't be blamed for the unreasonable use case (bug) - show details of an expense twice in one page.

**Comment by johncschuster** [comment]
This is still actively being discussed. This is just a bump for Melvin. Carry on!

**Comment by shawnborton** [comment]
Sorry - what is the exact question you need me to answer?

**Comment by johncschuster** [comment]
@rayane-djouah, bump on the above

**Comment by johncschuster** [comment]
@shawnborton ^^

**Comment by shawnborton** [comment]
Whatever you all think is best works for me. Just noting that a bunch of regressions came out of this one, and it seems like we have a new one that needs to be addressed here too: https://github.com/Expensify/App/issues/43247

**Comment by johncschuster** [comment]
Discussing payment. Will update the issue shortly.

**Comment by johncschuster** [comment]
I spoke with @yuwenmemon about this, and it sounds like there were two regressions within the 7-day waiting period: https://github.com/Expensify/App/issues/42760 and https://github.com/Expensify/App/pull/34640. Have I understood that correctly? Once I have a sense of the regressions caused, I'll update the payment summary so we can button this up.

Thanks for your patience everyone!

**Comment by johncschuster** [comment]
@rayane-djouah Ah, I did link the wrong one. Sorry about that!

Thank you for writing that out so clearly! I've spoken to the team, and we've agreed on the payment amounts originally dictated above. I'll copy them here for ease:

# Payment Summary:
Contributor: @rayane-djouah - ~$500~ $250 - paid [via Upwork](https://www.upwork.com/nx/wm/offer/100932868) on ~June 6~ TBD (will need to resolve regressions first)
Contributor+: @eh2077 - ~$500~ $250 - paid [via Upwork](https://www.upwork.com/nx/wm/offer/102574310) on ~June 6~ TBD (will need to resolve regressions first)
